aws/src/main/java/org/apache/iceberg/aws/glue/LakeFormationAwsClientFactory.java

```diff
@@ -0,0 +1,126 @@
+import software.amazon.awssdk.services.glue.GlueClient;
+import software.amazon.awssdk.services.kms.KmsClient;
+import software.amazon.awssdk.services.s3.S3Client;
+
+public class LakeFormationAwsClientFactory
+    implements AwsClientFactory {
+  private boolean useLakeFormationCredentials;
+  private Map<String, String> properties;
+  private Region region;
+  private String s3Endpoint;
+  private String accessKeyId;
+  private String secretAccessKey;
+  private String sessionToken;
+
+  /**
+   * No-arg constructor to load the catalog dynamically.
+   * * <p>
+   * All fields are initialized by calling {@link LakeFormationAwsClientFactory#initialize(Map)} later.
+   */
+  public LakeFormationAwsClientFactory() {
+  }
+
+  @Override
+  public void initialize(Map<String, String> catalogProperties) {
+    this.properties = Maps.newHashMap(catalogProperties);
+    if (properties.containsKey(LakeFormationCatalogProperties.USE_LAKE_FORMATION_CREDENTIALS) &&
+            properties.get(LakeFormationCatalogProperties.USE_LAKE_FORMATION_CREDENTIALS)
+                    .equals(String.valueOf(true))) {
+      useLakeFormationCredentials = true;
+    }
+    this.region = getRegion(properties);
+    this.s3Endpoint = properties.get(AwsProperties.S3FILEIO_ENDPOINT);
+  }
+
+  @Override
+  public S3Client s3() {
+    return S3Client.builder()
+            .endpointOverride(URI.create(s3Endpoint))
+            .credentialsProvider(buildS3CredentialProviderV2(useLakeFormationCredentials))
+            .region(region)
+            .httpClientBuilder(UrlConnectionHttpClient.builder())
+            .build();
+  }
+
+  @Override
+  public GlueClient glue() {
+    return GlueClient.builder()
+            .credentialsProvider(getAwsCredentialsProvider())
+            .region(region)
+            .httpClientBuilder(UrlConnectionHttpClient.builder())
+            .build();
+  }
+
+  @Override
+  public KmsClient kms() {
+    return KmsClient.builder()
+            .credentialsProvider(getAwsCredentialsProvider())
+            .region(region)
+            .httpClientBuilder(UrlConnectionHttpClient.builder())
+            .build();
+  }
+
+  @Override
+  public DynamoDbClient dynamo() {
+    return DynamoDbClient.builder()
+          .credentialsProvider(getAwsCredentialsProvider())
+          .region(region)
+          .httpClientBuilder(UrlConnectionHttpClient.builder())
+          .build();
+  }
+
+  private AwsCredentialsProvider buildS3CredentialProviderV2(boolean shouldUseLakeFormationCredentials) {
+    if (shouldUseLakeFormationCredentials) {
+      LakeFormationCredentialsProvider lakeFormationCredentialsProvider = new LakeFormationCredentialsProvider();
+      lakeFormationCredentialsProvider.initialize(properties);
+      return lakeFormationCredentialsProvider;
+    } else {
+      return getAwsCredentialsProvider();
+    }
+  }
+
+  private static Region getRegion(Map<String, String> properties) {
```

Comment:
Even for private methods, avoid using `get`. It is either unnecessary or should be replaced with a more specific verb.